packages/mui-system/src/cssVars/createCssVarsProvider.js

```diff
@@ -130,12 +131,22 @@ export default function createCssVarsProvider(options) {
     const styleSheet = {};
 
     Object.entries(colorSchemes).forEach(([key, scheme]) => {
-      const { css, vars } = cssVarsParser(scheme, {
+      const {
+        css,
+        vars,
+        parsedTheme: parsedScheme,
+      } = cssVarsParser(scheme, {
         prefix,
         basePrefix: designSystemPrefix,
         shouldSkipGeneratingVar,
       });
       mergedTheme.vars = deepmerge(mergedTheme.vars, vars);
+      if (key === resolvedColorScheme) {
+        mergedTheme = {
+          ...mergedTheme,
+          ...parsedScheme,
```

Comment:
Won't this override the `mergedTheme` palette values? Isn't the result of `parsedScheme` a object containing palette as the first key?

Reply:
ideally, `mergedTheme` should not have a palette as a key because the palette is handled inside color schemes.

Reply:
But if the original theme contains more values there, like functions in the MD theme, we will loose them again no?

Reply:
For example, this works but the functions need to be in all color schemes.

```js
createCssVarsProvider({
  theme: {
    colorSchemes: {
      light: {
        palette: {
          getContrastText: (...) => {},
        },
      },
      dark: {
        palette: {
          getContrastText: (...) => {},
        },
      }
    }
  }
})
```

Reply:
> ideally, `mergedTheme` should not have a palette as a key because the palette is handled inside color schemes.

I would then say we should scratch ~ideally~ :D The base theme cannot have the palette key, because it is auto generated by the `CssVarsProvider`.

Reply:
I will update my branch with this change

Reply:
Also, please document this, it's easy to be overlooked

Reply:
added.